pylint_exit_options.py

```diff
@@ -1,269 +1,302 @@
+        return [x[1] for x in self._decode(value)]
+
+    def _get_exit_code(self, value):
+        """Return the exist code that should be returned.
+
+        Args:
+            value(int): Return code from pylint command line.
+
+        Returns:
+            int: Return code that should be returned when run as a command.
+
+        Example:
+            >>> self._get_exit_code(1)
+            1
+            >>> self._get_exit_code(2)
+            2
+            >>> self._get_exit_code(3)
+            3
+            >>> self._get_exit_code(12)
+            4
+        """
+        exit_codes = [x[2] for x in self._decode(value)]
+        if not exit_codes:
+            return 0
+        return sum(exit_codes)
+
+    def handle_exit_code(self, value):
+        """
+        Exit code handler.
+
+        Takes a pylint exist code as the input parameter, and
+        displays all the relevant console messages.
+
+        Args:
+            value(int): Return code from pylint command line.
+
+        Returns:
+            int: Return code that should be returned when run as a command.
+
+        Example:
+            >>> ExitCodeMutator.handle_exit_code(1)
+            The following messages were raised:
+            <BLANKLINE>
+              - fatal message issued
+            <BLANKLINE>
+            Fatal messages detected.  Failing...
+            1
+            >>> ExitCodeMutator.handle_exit_code(12)
+            The following messages were raised:
+            <BLANKLINE>
+              - warning message issued
+              - refactor message issued
+            <BLANKLINE>
+            No fatal messages detected.  Exiting gracefully...
+            0
+        """
+        messages = self._get_messages(value)
+        exit_code = self._get_exit_code(value)
+
+        if messages:
+            print('The following types of issues were found:')
+            print('')
+
+            for message in messages:
+                print("  - %s" % message)
+
+            print('')
+
+        if exit_code:
+            print('The following types of issues are blocking:')
+            print('')
+
+            exit_messages = self._get_messages(exit_code)
+            for exit_message in exit_messages:
+                print("  - %s" % exit_message)
+
+            print('')
+
+        return exit_code
+
+    def _handle_cli_arg(self, namespace):
+        """Applies the CLI flags
+
+        Args:
+            namespace (argparse.Namespace): namespace from CLI arguments
+
+        """
+        if namespace.exit_report:
+            arg_value_list = namespace.exit_report.split(',')
+            self._set_report_arg_values(arg_value_list)
+
+    def _set_report_arg_values(self, arg_values: []):
+        """ Apply an enforcement setting
+
+        Args:
+            arg_values (List): A list of setting passed into the '--exit-report' option which can change which exit
+            codes will be returned to cli
+        """
+        if 'F' in arg_values:
```

Comment:
Why not use a `Letter => {PassSeverity, FailSeverity}` dictionary and loop through them?

Reply:
Thanks, I'll change this

Reply:
I reworked this the code looks like this now

"""
        check = {
            'F': __FATAL__,
            'E': __ERROR__,
            'W': __WARNING__,
            'R': __REFACTOR__,
            'C': __CONVENTION__,
            'U': __USAGE__
        }
        for key, value in check.items():
            if key in arg_values:
                self._apply_enforcement_setting(value, value)
            else:
                self._apply_enforcement_setting(value, __SUPPRESS__)
"""